src/engraving/rendering/dev/beamlayout.cpp

```diff
@@ -899,6 +904,12 @@ void BeamLayout::verticalAdjustBeamedRests(Rest* rest, Beam* beam, LayoutContext
     });
 
     Shape restShape = rest->shape().translated(rest->pagePos() - rest->offset());
+    if (up && firstRest) {
+        // Pad rest shape by small amount so horizontal distance between rest and beam is more than 0
+        // A horizontal distance of 0 counts as no vertical collision
+        RectF bbox = restShape.bbox();
+        restShape.addBBox(RectF(bbox.topRight(), PointF(bbox.bottomRight().x() + 1, bbox.bottomRight().y())));
```

Comment:
Instead of just `+ 1`, might it be desirable to add some spatium-based value? Or does it really not matter?

Reply:
@miiizen this is a good thing to do, but:

- Like Casper said, please use a spatium-based value for this padding and give it a descriptive name.
- This implementation (adding a rectangle to the Shape that's slightly larger than the Shape itself) is quite hacky. It would be better (and probably also useful in general) to change the verticalClearance method to something like `Shape::verticalClearance(otherShape, minHorizontalDistance)`, where minHorizontalDistance default-initializes to zero so it doesn't change where it's already used. Then, inside verticalClearance, you can add minHorizontalDistance to the left/right edges as needed before checking intersections :)